dcos_installer/test_backend.py

```diff
@@ -297,54 +274,50 @@ def test_do_aws_configure(tmpdir, monkeypatch):
         assert backend.do_aws_cf_configure() == 0
 
 
-valid_storage_config = """---
+@pytest.fixture
+def valid_storage_config(release_config_aws):
+    """ Uses the settings from dcos-release.config.yaml ['testing'] to create a
+    new upload and then deletes it when the test is over
+    """
+    s3_bucket_name = release_config_aws['bucket']
+    bucket_path = str(uuid.uuid4())
+    yield """---
 master_list:
  - 127.0.0.1
-aws_template_storage_access_key_id: {key_id}
 aws_template_storage_bucket: {bucket}
-aws_template_storage_bucket_path: mofo-the-gorilla
-aws_template_storage_secret_access_key: {access_key}
+aws_template_storage_bucket_path: {bucket_path}
 aws_template_upload: true
-"""
+""".format(
+        bucket=release_config_aws['bucket'],
+        bucket_path=bucket_path)
+    session = boto3.session.Session()
+    s3 = session.resource('s3')
+    s3_bucket = s3.Bucket(s3_bucket_name)
+    for o in s3_bucket.objects.filter(Prefix=bucket_path):
+        o.delete()
+    # objects = [{'Key': o.key} for o in s3_bucket.objects.filter(Prefix=bucket_path)]
+    # s3_bucket.delete_objects(Delete={'Objects': objects})
 
 
-def test_do_aws_cf_configure_valid_storage_config(config_aws, tmpdir, monkeypatch):
-    bucket = str(uuid.uuid4())
-    config_str = valid_storage_config.format(
-        key_id=config_aws["access_key_id"],
-        bucket=bucket,
-        access_key=config_aws["secret_access_key"])
-    assert aws_cf_configure(bucket, config_str, config_aws, tmpdir, monkeypatch) == 0
+def test_do_aws_cf_configure_valid_storage_config(release_config_aws, tmpdir, monkeypatch, valid_storage_config):
```

Comment:
same here with w.r.t order of args. We could consider keeping the patcher arg to the last one.